compiler/sempass2.nim

```diff
@@ -289,7 +289,7 @@ proc listSideEffects(result: var string; s: PSym; cycleCheck: var IntSet;
 
 proc listSideEffects(result: var string; s: PSym; conf: ConfigRef; context: PContext) =
   var cycleCheck = initIntSet()
-  result.addf "'$#' can have side effects\n" % s.name.s
+  result.addf("'$#' can have side effects\n", s.name.s)
```

Comment:
nice catch; we should probably issue a warning error when addf is used with varargs.len == 0

Reply:
Good idea. I'll submit a PR for it. Warning or error?

Reply:
warnings are more flexible because they can be turned into errors, but not vice versa, and it's easy to change whether it defaults as a warningAsError:warnSuspiciousAddf:on|off so it gives a natural migration path without breaking change

Reply:
A proc ident `$` was formatted into the first string and then that was passed to `strutils.addf`.

Reply:
It was invalid code because of side effects that encouraged me to make this PR.

Reply:
see also https://github.com/nim-lang/Nim/pull/14282 which may touch related code

Reply:
I don't know what I was thinking about putting a warning in there for this. It's not the compiler.
Is there a compile time method of checking a varargs' length? Should there be?
```
proc a(b: varargs[string]) =
  when b.len == 0: {.warning: "I need args!".}
  discard
```

Reply:
I just found `system.varargsLen` but it doesn't show the warning.

Reply:
`varargsLen` works inside templates, not procs, refs https://github.com/nim-lang/Nim/issues/18463

let's discuss this in https://github.com/timotheecour/Nim/issues/780 instead to avoid discussing something unrelated to this PR